### PR TITLE
Make connectivity with redis more reliable/robust.

### PR DESCRIPTION
**Pull request by kriegerse**
If redis fails try to reconnect, instead of asume connectivity is always available.

### CODE REVIEW

File: src/write_redis.c [hunk 1]
```diff
@@ -55,6 +55,39 @@ typedef struct wr_node_s wr_node_t;
...
+     return NULL;
+    }
+   else if (node->conn->err)
+   {
+     ERROR ("write_redis plugin: Connecting to host \"%s\" (port %i) failed: %s",
+             (node->host != NULL) ? node->host : "localhost",
+             (node->port != 0) ? node->port : 6379,
+             node->conn->errstr);
+     pthread_mutex_unlock (&node->lock);
+     return NULL;
```
Comment (mfournier) [inline src/write_redis.c]: Shouldn't you free `node->conn` here ?

File: src/write_redis.c [hunk 2]
```diff
@@ -55,6 +55,39 @@ typedef struct wr_node_s wr_node_t;
 /*
  * Functions
  */
+static redisContext * wr_connect ( wr_node_t *node ) /* {{{ */
+{
+  redisReply   *rr;
+
+  node->conn = redisConnectWithTimeout ((char *)node->host, node->port, node->timeout);
+   if (node->conn == NULL)
```
Comment (mfournier) [inline src/write_redis.c]: There's a little indentation issue here

File: src/write_redis.c (hunk 1 above)
Comment (kriegerse) [inline src/write_redis.c]: should be fixed with commit 51dee33f9c27b3030d5f8625f70977e41bb0e021

File: src/write_redis.c (hunk 2 above)
Comment (kriegerse) [inline src/write_redis.c]: should be fixed with commit 51dee33f9c27b3030d5f8625f70977e41bb0e021

**Comment by kriegerse** [comment]
write_redis and collectd now survives, if redis is not available or was restarted since the last write.

**Pull request closed without merging**